tests/platform_tests/test_process_reboot_cause.py

```diff
@@ -0,0 +1,190 @@
+    pytest.mark.device_type('physical'),
+]
+
+SYSTEMCTL_SHOW_CMD = "systemctl show process-reboot-cause.service --property='ExecMainPID' --property='ExecMainStatus' --property='Result'"  # noqa: E501
+SYSTEMCTL_RESTART_CMD = "systemctl restart process-reboot-cause.service"
+JOURNALCTL_CMD = "journalctl -u process-reboot-cause.service _PID={}"
+
+REBOOT_CAUSE_GEN_TIME_CMD = 'date +"%Y_%m_%d_%H_%M_%S"'
+REBOOT_CAUSE_HISTORY_DIR = "/host/reboot-cause/history/"
+REBOOT_CAUSE_FILE_FORMAT = "reboot-cause-{}.json"
+REBOOT_TYPES = ['warm-reboot', 'fast-reboot', 'soft-reboot', 'reboot', 'Power loss', 'Watchdog',  'Unknown', 'Hardware - Other', 'Non-Hardware']  # noqa: E501
+BAD_JSON = "{'gen_time:'"
+GOOD_JSON = {"gen_time": "9999_99_99_99_99_99", "cause": "", "user": "admin", "time": "N/A", "comment": "N/A"}  # noqa: E501
+JSON_DECODE_ERROR = "json.decoder.JSONDecodeError"
+EXCEPTION_HANDLED_SYSLOG = "Unable to process reload cause file"
+REBOOT_CAUSE_TEST_IDENTIFIER = "process-reboot-cause-test"
+
+
+class TestProcessRebootCause():
+
+    duthost = None
+    image_ver = None
+    image_major_ver = None
+    image_minor_ver = None
+
+    @pytest.fixture(scope="function", autouse=True)
+    def setup(self, duthosts, enum_rand_one_per_hwsku_hostname):
+        if self.duthost is None:
+            self.duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+        skip, reason = check_skip_release(self.duthost, ["201811", "201911"])
+        if skip is True:
+            pytest.skip("Skip test 'process-reboot-cause' for {} running image {} due to reason: {}".format(self.dthost.facts['platform'], self.duthost.os_version, reason))  # noqa: E501
+
+        # Get image version on DUT
+        self.image_ver = self.duthost.sonichost.os_version
+
+        # Get major and minor version of image
+        self.image_major_ver = int(self.image_ver.split('.')[0])
+        self.image_minor_ver = int(self.image_ver.split('.')[-1])
+
+    ##########################
+    #                        #
+    #      HELPER FNS        #
+    #                        #
+    ##########################
+
+    def get_service_status(self):
+        """
+        @summary: Check status of process-reboot-cause.service and return parameters to calling function  # noqa: E501
+        """
+
+        status = -1
+        main_pid = -1
+        result = ""
+
+        systemctl_output = self.duthost.command(SYSTEMCTL_SHOW_CMD)
+
+        for line in systemctl_output["stdout_lines"]:
+            if "ExecMainPID" in line:
+                main_pid = int(line.split("=")[-1])
+            elif "ExecMainStatus" in line:
+                status = int(line.split("=")[-1])
+            elif "Result" in line:
+                result = line.split("=")[-1]
+
+        return main_pid, status, result
+
+    def restart_service(self):
+        logging.info("Restarting process-reboot-cause.service")
+        _ = self.duthost.command(SYSTEMCTL_RESTART_CMD)
+
+    def create_json_file(self, filetype=""):
```

Comment:
@assrinivasan where do we cleanup these created json file after the test completes?

Reply:
I refactored the code -- now we clean up the JSON file in the runner function by calling remove_json_file() function, from the fixture that runs before and after each test. Please see attached sonic-mgmt logs